Products/ZPsycopgDA/db.py

```diff
@@ -362,12 +362,27 @@ def is_serialization_error(error):
             and 'could not serialize' in value
         )
 
+    @staticmethod
+    def is_spurious_collision_error(error):
+        '''
+        Detect whether the error condition is spurious or transitional and
+        should therefore be retried without delay.
+
+        So far, only Deadlock errors fall in this category.
+        '''
+        (name, value) = DB.split_error(error)
+        if name == 'DeadlockDetected':
+            return True
+        return False
 
     def handle_retry(self, error):
         '''Find out if an error deserves a retry.'''
         if self.is_serialization_error(error):
             raise RetryError from error
 
+        if self.is_spurious_collision_error(error):
+            raise RetryError
```

Comment:
No chaining here like `raise RetryError from error` ? Can't find direct reason to skip this here.